### ISSUE TITLE
-watchLifecycleEvents: is listening to the same notification for foreground/background.

### ISSUE THREAD

**Issue by bolsinga** [bugsnag/bugsnag-cocoa#521]
### Description
I was reading the code to understand more and found the issue I describe here.

### Issue
```
#if TARGET_OS_TV || TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE
    foregroundName = UIApplicationWillEnterForegroundNotification;
    backgroundName = UIApplicationWillEnterForegroundNotification;
    #elif TARGET_OS_MAC
    foregroundName = NSApplicationWillBecomeActiveNotification;
    backgroundName = NSApplicationDidFinishLaunchingNotification;
    #endif

```
### Environment

Library versions:

master

it should be
    `backgroundName = UIApplicationDidEnterBackgroundNotification;`

**Comment by xander-jones** [comment]
Hey @bolsinga, thanks for this report. We will take a look into this.

**Comment by xander-jones** [comment]
Hey @bolsinga, the fix for this has been released in [v5.23.1](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v5.23.1)
